### ISSUE TITLE
Memory leak on consumer.Close()

### ISSUE THREAD

**Issue by nickmurr** [apache/pulsar-client-go#355]
#### Expected behavior

Regarding this issue [154](https://github.com/apache/pulsar-client-go/pull/154) it seems that this problem still exists. 
I still get a memory leak on closing the consumer

When i do this (as example below) then the memory leaks happen

```go
for k, v := range TopicsRedis {
  consumerChannel := make(chan pulsar.ConsumerMessage, v.Priority)
  consumer, err := m.PulsarClient.Subscribe(pulsar.ConsumerOptions{
	Topic:                                    config.Vars.TopicPrefix + k,
	SubscriptionName:             "my-sub",
	Type:                                     pulsar.Shared,
	SubscriptionInitialPosition: pulsar.SubscriptionPositionEarliest,
	MessageChannel:                consumerChannel,
  })
 ...
consumer.Close()
```

#### System configuration
**Pulsar version**: 0.1.1

**Comment by vlemaitre** [comment]
Hello,
I faced the same issue using the library version 0.1.1.
Moving on version 0.2.0 solved the issue !

**Comment by wolfstudy** [comment]
The issue already fixed in 0.2.0, will close issue.
